src/order-book/types.ts

```diff
@@ -6,3 +6,44 @@ import { Order } from './generated'
 export interface EnrichedOrder extends Order {
   totalFee: string
 }
+
+/**
+ * CoW Protocol quote has amounts (sell/buy) and costs (network fee), there is also partner fees.
+ * Besides that, CoW Protocol supports both sell and buy orders and the fees and costs are calculated differently.
+ *
+ * The order of adding fees and costs is as follows:
+ * 1. Network fee is always added to the sell amount
+ * 2. Partner fee is added to the surplus amount (sell amount for sell-orders, buy amount for buy-orders)
+ *
+ * For sell-orders the partner fee is subtracted from the buy amount after network costs.
+ * For buy-orders the partner fee is added on top of the sell amount after network costs.
+ */
+export interface QuoteAmountsAndCosts {
+  isSell: boolean
+
+  quotePrice: number
```

Comment:
I have a concern regarding this being a number.
Will the precision here be a problem?

Reply:
This is the only way I see.
Yes, the price is a fraction and because of it it cannot be BigInt.

Reply:
I really liked the old library bigdecimal.js, it supported decimals with arbitrary precission. For interfaces maybe makes sense to make it a string and then anyone uses the tool they consider to part this to a decimal representation (or a number if they please)

Alternatively, not sure if it's too bad, but we could make it oppinionated library and marry to a library like bigdecimal.js so if people don't mind to import this into their project they have an easier time handling our data.

Reply:
The price is actually is not super useful in the result, I think it's better to remove it at all. Not a problem in any case to divide `buyAmount / sellAmount` on the consumer side

Reply:
Removed it

Reply:
okok